### PR TITLE
feat: add konfig-manager cli

### PR DESCRIPTION
**Pull request by Kaitou786**
Test the code with sample files present:
```
cat sample/data.yaml | go run main.go generate -A=spring,quarkus -c sample/config.yml
```
The above command will create spring.properties and quarkus.properties in the pwd.... You can specify the -o for the directory where these files will be created

### CODE REVIEW

**Comment by Kaitou786** [comment]
@moshloop in the second commit I have templated without any structs.. passing the string directly for that I had to use `{{ . }}` in the config.yml file... Not sure let me know which one you like :)

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+)
+
+var input string
+var output string
+var config string
+var applicationNames []string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "konfig-manager",
```
Comment (moshloop) [inline cmd/root.go]: can we move this into a generate sub command in cmd/generate.go

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			if err != nil {
+				return err
+			}
+		} else {
+			buf, err = ioutil.ReadFile(input)
+			if err != nil {
+				return err
+			}
+		}
+		resources, _ := kustomize.GetUnstructuredObjects(buf)
```
Comment (moshloop) [inline cmd/root.go]: move this to pkg/generate.go

File: cmd/root.go (hunk 1 above)
Comment (Kaitou786) [reply cmd/root.go]: created sub-command

File: pkg/properties_generator.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+func generatePropertiesForApp(input Input, output, applicationName string) error {
+	itemsMap := getObjectsFromHierarchy(input)
+	propertiesMap := make(map[string]interface{})
+	var properties string
+	for item, key := range itemsMap {
+		if key != "data" {
+			properties = properties + item.Object["data"].(map[string]interface{})[key].(string)
+		} else {
+			data := item.Object["data"].(map[string]interface{})
+			if item.Object["kind"].(string) == "Secret" {
```
Comment (moshloop) [inline pkg/properties_generator.go]:
```suggestion
			if item.GetKind() == "Secret" {
```

File: pkg/properties_generator.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+func generatePropertiesForApp(input Input, output, applicationName string) error {
+	itemsMap := getObjectsFromHierarchy(input)
+	propertiesMap := make(map[string]interface{})
+	var properties string
+	for item, key := range itemsMap {
+		if key != "data" {
+			properties = properties + item.Object["data"].(map[string]interface{})[key].(string)
+		} else {
+			data := item.Object["data"].(map[string]interface{})
```
Comment (moshloop) [inline pkg/properties_generator.go]:
```suggestion
			data :=unstructured.GetUnstructuredMap(item, "data")
```

File: pkg/properties_generator.go (hunk 3 above)
Comment (Kaitou786) [reply pkg/properties_generator.go]: Updated

File: pkg/properties_generator.go (hunk 4 above)
Comment (Kaitou786) [reply pkg/properties_generator.go]: Changed with `unstructured.NestedMap`

File: pkg/hierarchy.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	return nil, ""
+}
+
+func getHierarchyItems(configFile, applicationName string) ([]Item, error) {
+	var items []Item
+	buf, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return nil, err
+	}
+	config := make(map[string]interface{})
```
Comment (moshloop) [inline pkg/hierarchy.go]: This should be of type hiearachy

Comment (Kaitou786) [reply pkg/hierarchy.go]: changed

**Pull request merged**